### ISSUE TITLE
Creating a new Thorntail project fails - build failure due to quota being exceeded

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#4547]
##### Issue Overview
Creating a new project with the Thorntail runtime fails with build error.

##### Expected Behaviour
It should be possible for users to create new projects configured with the Thorntail runtime.

##### Current Behaviour
See build log fragment below for error.

##### Steps To Reproduce
1. Create new space and new project
2. For the project, select Mission = REST, and runtime = Thorntail
3. Observe build failure

##### Additional Information

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.openshift.booster.GreetingServiceTest
Resolving 395 out of 658 artifacts
Picked up _JAVA_OPTIONS: -Duser.home=/home/jenkins -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -Dsun.zip.disableMemoryMapping=true -XX:+UseParallelGC -XX:MinHeapFreeRatio=5 -XX:MaxHeapFreeRatio=10 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Xms10m -Xmx192m
Thu Nov 15 01:55:25 UTC 2018 INFO [org.wildfly.swarm.bootstrap] (main) Dependencies not bundled; resolving from M2REPO.
[0m2018-11-15 01:55:42,757 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:                   JAX-RS - STABLE          io.thorntail:jaxrs:2.2.0.Final
[0m[0m2018-11-15 01:55:42,832 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:                  Elytron - STABLE          io.thorntail:elytron:2.2.0.Final
[0m[0m2018-11-15 01:55:42,833 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:      MicroProfile Health - STABLE          io.thorntail:microprofile-health:2.2.0.Final
[0m[0m2018-11-15 01:55:42,833 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:                 Undertow - STABLE          io.thorntail:undertow:2.2.0.Final
[0m[0m2018-11-15 01:55:42,833 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:               Management - STABLE          io.thorntail:management:2.2.0.Final
[0m[0m2018-11-15 01:55:42,833 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:             Transactions - STABLE          io.thorntail:transactions:2.2.0.Final
[0m[0m2018-11-15 01:55:42,833 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:                      CDI - STABLE          io.thorntail:cdi:2.2.0.Final
[0m[0m2018-11-15 01:55:42,834 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:                  Logging - STABLE          io.thorntail:logging:2.2.0.Final
[0m[0m2018-11-15 01:55:42,834 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:      MicroProfile Config - STABLE          io.thorntail:microprofile-config:2.2.0.Final
[0m[0m2018-11-15 01:55:42,834 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:          Bean Validation - STABLE          io.thorntail:bean-validation:2.2.0.Final
[0m[0m2018-11-15 01:55:42,834 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:               Arquillian - STABLE          io.thorntail:arquillian:2.2.0.Final
[0m[0m2018-11-15 01:55:42,834 INFO  [org.wildfly.swarm] (main) THORN0013: Installed fraction:        CDI Configuration - STABLE          io.thorntail:cdi-config:2.2.0.Final
/opt/rh/rh-maven33/root/usr/bin/mvn: line 9:   158 Killed                  $M2_HOME/bin/mvn "$@"
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // container
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // podTemplate
[Pipeline] End of Pipeline

GitHub has been notified of this commit’s build result

ERROR: script returned exit code 137
Finished: FAILURE
```

**Comment by ldimaggi** [comment]
This is in production - cannot yet recreate on prod-preview as env reset results in Jenkins pod not starting.

**Comment by ldimaggi** [comment]
Seeing this on prod-preview too:

![screenshot from 2018-11-15 08-31-51](https://user-images.githubusercontent.com/642621/48556096-f2c9ce00-e8b0-11e8-99f8-8816e1424ed6.png)

**Comment by chmouel** [comment]
You seem to have run with an infamous [OOM](https://www.kernel.org/doc/gorman/html/understand/understand016.html) which basically mean your software is consuming too much memory to run on starter's OSIO restricted environment,  

if you can it would be nice if you can copy and paste the `oc get ev` from the -jenkins namespace  to confirm this,

**Comment by ldimaggi** [comment]
Your guess is correct - see log fragment below.

Note that the first step in the test was to reset the user's OpenShift.io env - no other spaces or projects were present when the test was run:

```
18m        18m         1         jenkins-1                   ReplicationController                                 
Warning   FailedCreate                  {replication-controller }                               
Error creating: pods "jenkins-1-crd5p" is forbidden: exceeded quota: compute-resources, 
requested: limits.cpu=2,limits.memory=1Gi, used: limits.cpu=2,limits.memory=1Gi, 
limited: limits.cpu=2,limits.memory=1Gi
18m 
```

**Comment by alexeykazakov** [comment]
If some booster project can’t be built on OSO then it should be removed from the wizard. This is something for the getting started team.

I’m reopening the issue.

**Comment by bartoszmajsak** [comment]
Is removal the right thing to do? How about we try to optimize the booster itself? @slemeur any comment on that?

@ldimaggi I have two questions here

- is that a permanent issue?
- is it the only affected booster for this runtime?

**Comment by slemeur** [comment]
Yes, if it was working before, it's a regression and we need to look at that.

**Comment by GeorgeActon** [comment]
@bartoszmajsak Who will look at the booster.

**Comment by bartoszmajsak** [comment]
@GeorgeActon App Creation aka Getting Started with help of booster owner if our initial analysis will lead to a conclusion that we have to improve booster to make it running in our env

**Comment by GeorgeActon** [comment]
@sbose78 All indications at this point are to look at the Booster. Please sync with @chmouel and figure next steps.

**Comment by ldimaggi** [comment]
Yes - this is a regression. It is 100% reproducible. And it blocks the user from creating the booster. As there is no workaround I think that it is a SEV1. Thx!

**Comment by chmouel** [comment]
I think the first thing would be to look at the booster and see if there was any update for the last couple of weeks and track down the regression between the versions,  I am not sure what is the girhub repo but assuming is this one there was a version update https://github.com/thorntail-examples/health-check/commit/6291567de288d4ed52aa908c21b3b163c2c641aa

**Comment by GeorgeActon** [comment]
It fails for either REST API and for Health check. tried both this morning.

**Comment by bartoszmajsak** [comment]
This is how boosters are mapped:

- defined in the catalog (and enabled for OSIO) https://github.com/fabric8-launcher/launcher-booster-catalog/blob/master/thorntail/community/rest-http/booster.yaml#L10
- pointing to version 15 (tag) https://github.com/thorntail-examples/rest-http/tree/15 (which has been updated month ago)

**Comment by sbose78** [comment]
I agree @bartoszmajsak . @corinnekrych , let's take a look at possibly fixing the booster. 

Btw, If that doesn't happen soon enough, we should temporarily take the booster out and then put it back only after its fixed. You OK with that, @slemeur ?

**Comment by slemeur** [comment]
Yes that would be fine with me @sbose78

**Comment by ldimaggi** [comment]
+1 to removing the booster until it works - anything visible should be working.

**Comment by corinnekrych** [comment]
Let's start with disabling it see PR.
@bartoszmajsak do you ping dev directly or automatic reviewer `launcher-devs` is enough?

**Comment by sbose78** [comment]
FYI: Figuring out why something which was working is now breaking, and not owned by us, will take a while. The sev1 might take longer to resolve.

**Comment by chmouel** [comment]
I vote for this boost being only available when building on a OSD clusters, it may be possible to fit it to run on OSO but the users would never being able to use it :(

**Comment by corinnekrych** [comment]
I've tested v11
and it's failed too. @ldimaggi so [v11](https://github.com/thorntail-examples/health-check/tree/11) (last updated  24th July) and [v12](https://github.com/thorntail-examples/health-check/tree/12) (last updated 16th October) of the thorntail healtg-check app and both [fails](https://jenkins.prod-preview.openshift.io/job/corinnekrych/job/ewewewewe/job/master/1/console) on my prod-preview => which confirm that this is not a regression in booster themselves but indeed as @chmouel mention in the ticket an issue with build.
I'll add that to the ticket. cc @shoubhikbose I'll remove my name from the ticket as I think this ticket will be better handle in build team. Let me know if you need more help on my side.

**Comment by sbose78** [comment]
Thanks for the investigation @corinnekrych and @chmouel . 

> This may get better tho when we move over the new osio-pipeline library, since we will not be building anymore in the -jenkins namespace but in user namespace,

I'll give this a shot.

**Comment by chmouel** [comment]
Thanks @sbose78 this would be a nice use case for osio-pipeline,

**Comment by sbose78** [comment]
Both the health-check and rest-http booster works with the new pipeline library since the build happens in the user namespace.

Example:
https://github.com/thorntail-examples/rest-http/pull/53
https://github.com/fabric8io/fabric8-jenkinsfile-library/pull/44/

Once we finish migrating our Java boosters to use the new pipeline library, then we can update the `Jenkinsfile`s and this should just work. 

@rupalibehera let's work together on getting the migration done.

**Comment by sthaha** [comment]
@sbose78 

> @rupalibehera let's work together on getting the migration done.

https://github.com/openshiftio/openshift.io/issues/4535 would need to be done before we can start using the new pipeline for java projects.

**Comment by gorkem** [comment]
#4535 is not a blocker for migrating **only** thorntail, please go ahead and enable thorntail without waiting for it. #4535 is a blocker for migrating all other boosters.
